crates/ruff_linter/src/rules/pycodestyle/overlong.rs

```diff
@@ -0,0 +1,142 @@
+use std::borrow::Cow;
+
+use unicode_width::UnicodeWidthStr;
+
+use ruff_python_index::Indexer;
+use ruff_python_trivia::is_pragma_comment;
+use ruff_source_file::Line;
+use ruff_text_size::{TextLen, TextRange};
+
+use crate::line_width::{LineLength, LineWidthBuilder, TabSize};
+
+#[derive(Debug)]
+pub(super) struct Overlong {
+    range: TextRange,
+    width: usize,
+}
+
+impl Overlong {
+    /// Returns an [`Overlong`] if the measured line exceeds the configured line length, or `None`
+    /// otherwise.
+    pub(super) fn try_from_line(
+        line: &Line,
+        indexer: &Indexer,
+        limit: LineLength,
+        task_tags: &[String],
+        tab_size: TabSize,
+    ) -> Option<Self> {
+        // The maximum width of the line is the number of bytes multiplied by the tab size (the
+        // worst-case scenario is that the line is all tabs). If the maximum width is less than the
+        // limit, then the line is not overlong.
```

Comment:
That's very conservative (but correct). It means only lines with less than 22 characters skip the check below. This optimisation becomes less effective if we increase the default `tab_size` to 8 (making it only applies to lines with less or equal to 11 characters). 

Although I'm not sure what we should do about it. Dividing by two could be an option, considering that it's unlikely that every second character is a tab but it will reduce correctness,

Reply:
This was already present, just moved. We used to have a much better heuristic, because we assumed each byte was at most one character, but tabs ruin that -- a single byte can be as large as the tab width, sadly.